posts/2020-09-01-jakarta-microprofile-beta-20.0.0.10.adoc

```diff
@@ -0,0 +1,336 @@
+
+Fault Tolerance 3.0 overhauls the metrics automatically exported by Fault Tolerance to take advantage of metric tags to make it easier to use those metrics to understand when and where your application is failing.
+
+Information which was previously included in the metric name is now included as metric tags, making it much easier to query for data from multiple methods and pick out which, if any, is causing issues.
+
+Example:
+
+Old metric: `application:ft.<name>.timeout.callsTimedOut.total`
+
+New metric: `base:ft.timeout.calls.total{method="<name>", timedOut="true"}``
+
+
+==== Try it now 
+
+Enable Fault Tolerance 3.0 and CDI in the server.xml, along with any other features you're using.
+
+[source, xml]
+----
+<featureManager>
+  <feature>mpFaultTolerance-3.0</feature>
+  <feature>cdi-2.0</feature>
+  <feature>jaxrs-2.1</feature>
+</featureManager>
+----
+
+Add any of the Fault Tolerance annotations to a business method of a CDI bean.
+
+[source, java]
+----
+@Retry(5)
+public User lookupUserFromRegistry(String name) {
+  return registry.getUser(name);
+}
+----
+
+When that method is called as a a CDI business method, the fault tolerance strategy will be used. In this example, if `lookupUserFromRegistry()` would throw an exception, it will instead be retried up to five times.
+
+Get an introduction to MicroProfile Fault Tolerance with the Open Liberty guides:
+
+* link:https://openliberty.io/guides/retry-timeout.html[Failing fast and recovering from errors]
+* link:https://openliberty.io/guides/circuit-breaker.html[Preventing repeared failed calls to microservices]
+
+References to:
+
+* link:https://download.eclipse.org/microprofile/microprofile-fault-tolerance-3.0-RC2/apidocs/[Java]
+
+* link:https://download.eclipse.org/microprofile/microprofile-fault-tolerance-3.0-RC2/microprofile-fault-tolerance-spec.html[Full Specification]
+
+* link:https://download.eclipse.org/microprofile/microprofile-fault-tolerance-3.0-RC2/microprofile-fault-tolerance-spec.html#_integration_with_microprofile_metrics[New format for metrics in detail in Chapter 11]
+
+link:https://github.com/OpenLiberty/open-liberty/[Report any issues on Github]
+
+
+[#metric]
+=== MicroProfile Metric 3.0
+
+MicroProfile Metrics 3.0 (as part of MicroProfile 4.0) introduces new metric values for the existing SimpleTimer and Timer metrics. Additionally a new REST metric is introduced for better monitoring and handling of unmapped exceptions. Manual configuration for re-usability has been removed. A notable change to the MP Metrics programming model regarding CDI Producers has been made. Lastly a medley of API improvements and refactoring have been added in this release.
+
+==== SimpleTimer
+The SimpleTimer metric now tracks and reports the highest and lowest recorded time duration of the previous complete minute. See the SimpleTimer.java for more more information in the Javadoc.
```

Comment:
https://download.eclipse.org/microprofile/microprofile-metrics-3.0-RC1/apidocs is this one okay?

Reply:
No, I think be specific because the javadoc is mentioned in each section.
For SimpleTimer: https://download.eclipse.org/microprofile/microprofile-metrics-3.0-RC1/apidocs/
For Timer: https://download.eclipse.org/microprofile/microprofile-metrics-3.0-RC1/apidocs/
(I assume those are the pages intended by the author.)

Can you also remove `.java` from each of those two sections - that's a really weird way to refer to javadoc? (at least, it is when there's a direct link to it hosted online, which we should always do anyway in blog posts etc)